src/odemis/odemisd/mdupdater.py

```diff
@@ -355,6 +355,21 @@ def observeFilter(self, filter, comp_affected):
         def updateOutWLRange(pos, fl=filter, comp_affected=comp_affected):
             wl_out = fl.axes["band"].choices[pos["band"]]
             comp_affected.updateMetadata({model.MD_OUT_WL: wl_out})
+            # apply lateral chromatic correction to align with the reference channel
+            apply_transform = fl.getMetadata().get(model.MD_CHROMATIC_COR, None)
+            if apply_transform:
+                try:
+                    metadata_cor = apply_transform[fl.position.value["band"]]
+                    assert type(metadata_cor) == dict
+                    assert all(isinstance(key, str) for key in metadata_cor)
+                    comp_affected.updateMetadata(metadata_cor)
+                except (AssertionError, KeyError) as exp:
+                    logging.error(
+                        "Chromatic correction metadata was not updated due to %s. Check if the CHROMATIC_COR is a "
```

Comment:
Passing the exception error message here doesn't work well currently because the "assert" raise an AssertionError with no message. => Add a message to the `assert` or use a `if ...: raise ValueError(...)`.